node/metrics.go

```diff
@@ -110,7 +110,7 @@ func makeWSMetrics() jsonrpc.NewRequestListener {
 	}
 }
 
-func makeRPCMetrics(version, legacyVersion string) (jsonrpc.EventListener, jsonrpc.EventListener) {
+func makeRPCMetrics(version1, version2, version3 string) (jsonrpc.EventListener, jsonrpc.EventListener, jsonrpc.EventListener) {
```

Comment:
Is it possible to isolate this function? In the future, the number of versions may differ.
```suggestion
func makeRPCMetrics(version string) jsonrpc.EventListener {
```

Reply:
I think this is a better approach to what I suggested below

Reply:
I reverted the change to this method, calling it multiple times causes
`prometheus.MustRegister(requests, failedRequests, requestLatencies)`
to be called multiple times, resulting in a panic.

Reply:
Cleaning how we handle rpcServers and registration etc could be a good first issue imo